packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -229,10 +230,27 @@ class MDnsVmServiceDiscovery {
       final Set<String> uniqueDomainNamesInResults = <String>{};
 
       // Listen for mDNS connections until timeout.
-      final Stream<PtrResourceRecord> ptrResourceStream = client.lookup<PtrResourceRecord>(
-        ResourceRecordQuery.serverPointer(dartVmServiceName),
-        timeout: timeout
-      );
+      final Stream<PtrResourceRecord> ptrResourceStream;
+
+      try {
+        ptrResourceStream = client.lookup<PtrResourceRecord>(
+          ResourceRecordQuery.serverPointer(dartVmServiceName),
+          timeout: timeout,
+        );
+      } on SocketException catch (e, stacktrace) {
+        _logger.printError(e.message);
+        _logger.printTrace(stacktrace.toString());
+        if (globals.platform.isMacOS) {
+          throwToolExit(
+              'You might be having a permissions issue with your IDE. '
+              'Please try going to '
+              'System Settings -> Privacy & Security -> Local Network -> '
+              '[Find your IDE] -> Toggle ON, then restart your phone.');
```

Comment:
Style nit:

```suggestion
          throwToolExit(
            'You might be having a permissions issue with your IDE. '
            'Please try going to '
            'System Settings -> Privacy & Security -> Local Network -> '
            '[Find your IDE] -> Toggle ON, then restart your phone.',
          );
```

Reply:
wtf. I think my copy pasting is not perserving the spaces correctly.

Reply:
You might have autoformat on, you should disable it for the Flutter repo (but you might want to keep it for the packages repo)